org.thingpedia.iot.tv/dataset.tt

```diff
@@ -0,0 +1,217 @@
+dataset @org.thingpedia.iot.tv language "en" {
+
+    program := now => @org.thingpedia.iot.tv.state() => notify
+    #_[utterances=["is my tv on?",
+                   "check if my television is off",
+                   "is the tv turned on?",
+                   "check if my tv is switched off",
+                   "check if my television is turned on or off",
+                   "check if my tv is switched on or switched off"]];
+
+    query := @org.thingpedia.iot.tv.state()
+    #_[utterances=["if my television is on or off",
+                   "whether my tv is switched on or switched off"]];
+
+    stream := monitor @org.thingpedia.iot.tv.state()
+    #_[utterances=["when my television changes state"]];
+
+    stream (p_state : Enum(on, off)) := edge( monitor @org.thingpedia.iot.tv.state()) on (state == p_state)
+    #_[utterances=["when my tv turns ${p_state}",
+                   "if my television switches ${p_state}",
+                   "when the tv becomes ${p_state}",
+                   "if my television is ${p_state}",
+                   "when my tv changes to ${p_state}"]];
+
+    stream := edge( monitor @org.thingpedia.iot.tv.state()) on (state == enum(on))
+    #_[utterances=["when my tv turns on",
+                   "when I switch on the television",
+                   "if the television becomes on",
+                   "when my tv gets switched to on",
+                   "if my television is on",
+                   "when my television is turned on",
+                   "the moment my tv is switched on"]];
+
+    stream := edge( monitor @org.thingpedia.iot.tv.state()) on (state == enum(off))
+    #_[utterances=["when my tv is turned off",
+                   "if my television switches off",
+                   "the moment my tv becomes off",
+                   "if my television is off",
+                   "when my tv gets turned off"]];
+
+    action (p_power :Enum(on,off)) := @org.thingpedia.iot.tv.set_power(power=p_power)
+    #_[utterances=["turn ${p_power} my television",
+                   "switch ${p_power} the television",
+                   "${p_power} the tv",
+                   "turn ${p_power} my tv",
+                   "set the power on my tv to ${p_power}"]];
+
+    action := @org.thingpedia.iot.tv.set_power(power=enum(on))
+    #_[utterances=["on my television",
```

Comment:
We already noticed that "on my television" is a bad template... on/off are not verbs

Reply:
Well... technically wouldn't we want Almond to recognize when a user says "on the tv"?

Reply:
No, I don't think so. "on the tv" means "on top of the tv" which is not a command. Nobody will say that.

Reply:
Removed all "on/off my xxx" utterances in tv, speaker, vacuum, switch devices.